File: onion-architecture/java/src/main/java/com/example/infrastructure/config/DomainConfiguration.java
package com.example.infrastructure.config;

import com.example.application.EnrolmentApplicationService;
import com.example.application.ports.CourseRepository;
import com.example.application.ports.StudentRepository;
import com.example.domain.services.EnrolmentService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class DomainConfiguration {

    @Bean
    public EnrolmentService enrolmentService() {
        return new EnrolmentService();
    }
    
    @Bean
    public EnrolmentApplicationService enrolmentApplicationService(
        StudentRepository studentRepo,
        CourseRepository courseRepo,
        EnrolmentService enrolmentService) {
        return new EnrolmentApplicationService(studentRepo, courseRepo, enrolmentService);
    }
}

File: onion-architecture/kotlin/src/main/kotlin/com/example/infrastructure/config/DomainConfiguration.kt
package com.example.infrastructure.config

import com.example.application.EnrolmentApplicationService
import com.example.application.ports.CourseRepository
import com.example.application.ports.StudentRepository
import com.example.domain.services.EnrolmentService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class DomainConfiguration {

    @Bean
    fun enrolmentService(): EnrolmentService {
        return EnrolmentService()
    }

    @Bean
    fun enrolmentApplicationService(
        studentRepo: StudentRepository,
        courseRepo: CourseRepository,
        enrolmentService: EnrolmentService
    ): EnrolmentApplicationService {
        return EnrolmentApplicationService(studentRepo, courseRepo, enrolmentService)
    }
}